### PR TITLE
Issue 120 - Add image button and Image Search Bar

### PR DESCRIPTION
**Pull request by semnim**
related issue #120

fixed PR to merge into dev branch

Feature addition: "add new image button", additionally to the right click option of adding a new image.
grafik

Tested on Firefox Nightly Version 107.0a1 (2022-10-01) (64-Bit)

### CODE REVIEW

**Review by Ari24-cb24 — APPROVED** [review]
The only thing I notice is that the toolbar is now somewhat empty (I dislike the empty space there). I think a searchbar might be a good element to add to it. You want to add it as well?

**Comment by semnim** [comment]
Will do in the next days

**Comment by Ari24-cb24** [comment]
I just noticed, it seems like your branch originated from master. Try to avoid that in the future as we mainly use dev for development and master for releases 😅

**Comment by semnim** [comment]
will fix this in the upcoming PR

**Comment by semnim** [comment]
Edit on this PR:

- is now based off of dev branch
- implemented search bar


**example:**
before search
grafik

**after search**
grafik

File: src/components/settings/playlist_settings/playlistsettingscomponent.module.css [hunk 1]
```diff
@@ -37,3 +37,31 @@
...
+	background-color: transparent;
+	border-radius: 3px;
+	box-sizing: border-box;
+	border: 1px solid #4a4a4a;
+	font-size: 16px;
+	font-weight: 400;
+	color: #4a4a4a;
+	padding: 18px;
+	box-shadow: 0 0 15px rgba(0, 0, 0, 0.1);
+}
```
Comment (Ari24-cb24) [inline src/components/settings/playlist_settings/playlistsettingscomponent.module.css:66]: Currently there is a blue outline when the input is being focussed. If possible, change this to `outline: none` for now as a general color for all outlines isn't implemented yet.

File: src/utils/use-search.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from "react";
+import { IFolder, IImage, metaDb, ROOT_FOLDER } from "./db";
+
+export const useSearch = () => {
```
Comment (Ari24-cb24) [inline src/utils/use-search.ts:4]: If possible, rename this function and the file to something more descriptive e.g. use-image-search

File: src/components/settings/playlist_settings/PlaylistSettingsComponent.tsx [hunk 3]
```diff
@@ -333,36 +363,55 @@ function PlaylistSettingsComponent(props: { bodyRef: any }) {
...
+													.getImages(currentFolder.id)
+													.then(setImages),
+											50
+										);
+									}}
+									key={folder.id}
+								/>
+							);
+						}
+					})}
```
Comment (Ari24-cb24) [inline src/components/settings/playlist_settings/PlaylistSettingsComponent.tsx:388]: For sake of complying with react guidelines, add a default `null` return value

File: src/components/settings/playlist_settings/PlaylistSettingsComponent.tsx [hunk 4]
```diff
@@ -333,36 +363,55 @@ function PlaylistSettingsComponent(props: { bodyRef: any }) {
...
+								<Background
+									selected={currentImageId === image.id}
+									image={image}
+									index={index}
+									setDraggedElement={setDraggedElement}
+									key={index}
+								/>
+							);
+						}
+					})}
```
Comment (Ari24-cb24) [inline src/components/settings/playlist_settings/PlaylistSettingsComponent.tsx:401]: For sake of complying with react guidelines, add a default `null` return value

File: src/components/settings/playlist_settings/PlaylistSettingsComponent.tsx [hunk 5]
```diff
@@ -333,36 +363,55 @@ function PlaylistSettingsComponent(props: { bodyRef: any }) {
...
+									<Background
+										selected={false}
+										image={image}
+										index={index}
+										setDraggedElement={setDraggedElement}
+										key={index}
+									/>
+								}
+							</>
 						);
```
Comment (Ari24-cb24) [inline src/components/settings/playlist_settings/PlaylistSettingsComponent.tsx:415]: Why do you use Fragments (`<></>`) and the curley braces here? I don't think they are necessary. If possible, remove the return statement and change the map to an implicit arrow function with brackets.

**Review by Ari24-cb24 — CHANGES_REQUESTED** [review]
I really appreciate the effort you did there. There are some small things that should be changed but the main concepts are well-implemented. I also like the recursive search idea as my first thought was just to search throught the current folder.

**Comment by semnim** [comment]
Thanks for the feedback, I've implemented the suggestions.

**Review by Ari24-cb24 — APPROVED** [review]
Sorry for the late response but thanks for the work! Will merge in a second

**Pull request merged**